csbuild/_build/__init__.py

```diff
@@ -822,6 +850,36 @@ def Run():
 		with csbuild.Target("fastdebug"):
 			pass
 
+		if args.generate_solution:
+			shared_globals.runMode = csbuild.RunMode.GenerateSolution
+			if args.solution_path:
+				shared_globals.solutionPath = args.solutionPath
+			else:
+				shared_globals.solutionPath = "./Solutions/{}/".format(args.generate_solution)
+
+			shared_globals.solutionPath = os.path.abspath(shared_globals.solutionPath)
+
+			if args.solution_name:
+				solutionName = args.solution_name
+			else:
+				solutionName = "csbuild"
+
+			@csbuild.OnBuildFinished
+			def OnBuildFinished(projectList): # pylint: disable=unused-variable
+				"""
+				:param projectList: list of projects
+				:type projectList: list[project.Project]
+				"""
+				inputs = {}
+				solutionTool = shared_globals.allGenerators[args.generate_solution].solutionTool
```

Comment:
Shouldn't you be accessing the generator in a way that lets us check for `None`?  If the user passes in a generator name that isn't registered, outputting an error message is much friendlier and more informative to users than an exception callstack.